packages/cli/src/modules/insights/database/repositories/insights-by-period.repository.ts

```diff
@@ -228,4 +254,89 @@ export class InsightsByPeriodRepository extends Repository<InsightsByPeriod> {
+
+		const [sortField, sortOrder] = this.parseSortingParams(sortBy);
+		const sumOfExecutions = sql`SUM(CASE WHEN insights.type IN (${TypeToNumber.success.toString()}, ${TypeToNumber.failure.toString()}) THEN value ELSE 0 END)`;
+
+		const rawRowsQuery = this.createQueryBuilder('insights')
+			.select([
+				'metadata.workflowId AS "workflowId"',
+				'metadata.workflowName AS "workflowName"',
+				'metadata.projectId AS "projectId"',
+				'metadata.projectName AS "projectName"',
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.success} THEN value ELSE 0 END) AS "succeeded"`,
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.failure} THEN value ELSE 0 END) AS "failed"`,
+				`SUM(CASE WHEN insights.type IN (${TypeToNumber.success}, ${TypeToNumber.failure}) THEN value ELSE 0 END) AS "total"`,
+				sql`CASE
+								WHEN ${sumOfExecutions} = 0 THEN 0
+								ELSE 1.0 * SUM(CASE WHEN insights.type = ${TypeToNumber.failure.toString()} THEN value ELSE 0 END) / ${sumOfExecutions}
+							END AS "failureRate"`,
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.runtime_ms} THEN value ELSE 0 END) AS "runTime"`,
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.time_saved_min} THEN value ELSE 0 END) AS "timeSaved"`,
+				sql`CASE
+								WHEN ${sumOfExecutions} = 0	THEN 0
+								ELSE 1.0 * SUM(CASE WHEN insights.type = ${TypeToNumber.runtime_ms.toString()} THEN value ELSE 0 END) / ${sumOfExecutions}
+							END AS "averageRunTime"`,
+			])
+			.innerJoin('insights.metadata', 'metadata')
+			.where(`insights.periodStart >= ${dateSubQuery}`)
+			.groupBy('metadata.workflowId')
+			.addGroupBy('metadata.workflowName')
+			.addGroupBy('metadata.projectId')
+			.addGroupBy('metadata.projectName')
+			.orderBy(this.escapeField(sortField), sortOrder);
+
+		const count = (await rawRowsQuery.getRawMany()).length;
+		const rawRows = await rawRowsQuery.offset(skip).limit(take).getRawMany();
+
+		return { count, rows: aggregatedInsightsByWorkflowParser.parse(rawRows) };
+	}
+
+	async getInsightsByTime(nbDays: number) {
+		const dateSubQuery =
+			dbType === 'sqlite'
+				? `datetime('now', '-${nbDays} days')`
+				: dbType === 'postgresdb'
+					? `CURRENT_DATE - INTERVAL '${nbDays} days'`
+					: `DATE_SUB(CURDATE(), INTERVAL ${nbDays} DAY)`;
+
+		const rawRowsQuery = this.createQueryBuilder('insights')
+			.select([
+				'insights.periodStart AS "periodStart"',
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.runtime_ms} THEN value ELSE 0 END) AS "runTime"`,
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.success} THEN value ELSE 0 END) AS "succeeded"`,
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.failure} THEN value ELSE 0 END) AS "failed"`,
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.time_saved_min} THEN value ELSE 0 END) AS "timeSaved"`,
+			])
+			.where(`insights.periodStart >= ${dateSubQuery}`)
+			.addGroupBy('insights.periodStart') // TODO: group by specific time scale (start with day)
+			.orderBy('insights.periodStart', 'ASC');
```

Comment:
The default sorting is not covered by a test. Not sure if it needs be though. 🤔

Reply:
In the test we expect the result to be ordered by ascending date: https://github.com/n8n-io/n8n/blob/pay-2584-implement-be-api-for-insights-data/packages/cli/src/modules/insights/__tests__/insights.service.test.ts#L1097